src/AggregateFunctions/AggregateFunctionFactory.cpp

```diff
@@ -59,6 +59,34 @@ void AggregateFunctionFactory::registerFunction(const String & name, Value creat
     }
 }
 
+void AggregateFunctionFactory::registerAliasForAggregateFunctionWithCombinator(
```

Comment:
This PR is almost perfect!

But I was stuck here while reading this function.
It is only necessary to use the `isAggregateFunctionName` to check for correctness. And it imposes a restriction - the combinator should be already registered. But in all other places, it is copy-paste of `registerAlias`. What if we add `registerAliasUnchecked` to `IFactoryWithAliases`, and then all the complexity will vanish? (For example, we will remove `friend` and `Must register after registerCombinator`).

Reply:
In this way, does it means we don't check the `real_name` is a valid aggregate function in `registerAliasUnchecked` anymore?